### PR TITLE
Net app vx works64

### PR DESCRIPTION
**Pull request by shuston**
This is a set of changes I am assisting a customer with. I do not have a build environment to test with, but the customer does and they've approved the changes for VxWorks.

### CODE REVIEW

**Comment by jwillemsen** [comment]
Steve, I don't have a VxWorks environment but when I quickly look at the changes I think your pull request will break support for any VxWorks version < 6.9, for example _Vx_usr_arg_t doesn't seem to be supported in older releases, see ace/Stack_Trace.cpp where there are some ACE macros to workaround the unavailability of that type in older releases. Probably this is the case for any new type/define you use, but hard to say without someone setting up a full build with VxWorks 6.9 and one with an older release, maybe your customer can do that?

**Comment by shuston** [comment]
Good points, Johnny. I will work on the needed changes. Thanks for the quick review!

**Comment by jwillemsen** [comment]
The VxWorks 5.5 taskLib documentation describes the API of the older systems, see the following link for that old API description http://www.vxdev.com/docs/vx55man/vxworks/ref/taskLib.html

File: ACE/ace/OS_NS_Thread.cpp [hunk 1]
```diff
@@ -4472,7 +4472,7 @@ ACE_OS::thr_create (ACE_THR_FUNC func,
     }
 #   endif /* 0 */
 
-  if (tid == ERROR)
+  if (tid == TASK_ID_ERROR)
```
Comment (jwillemsen) [inline ACE/ace/OS_NS_Thread.cpp]: According to VxWorks 5.5 the return value in older VxWorks versions is really ERROR, not TASK_ID_ERROR, see http://www.vxdev.com/docs/vx55man/vxworks/ref/taskLib.html#taskSpawn

File: ACE/ace/OS_NS_Thread.cpp [hunk 2]
```diff
@@ -5292,19 +5298,19 @@ spae (FUNCPTR entry, ...)
...
   // larger stack size (instead of ::sp ()'s 20000).
-  int const ret = ::taskSpawn (argv[0],    // task name
-                               100,        // task priority
-                               VX_FP_TASK, // task options
-                               ACE_NEEDS_HUGE_THREAD_STACKSIZE, // stack size
-                               entry,      // entry point
-                               argc,       // first argument to main ()
-                               (int) argv, // second argument to main ()
-                               0, 0, 0, 0, 0, 0, 0, 0);
+  TASK_ID const ret = ::taskSpawn (argv[0],    // task name
```
Comment (jwillemsen) [inline ACE/ace/OS_NS_Thread.cpp]: Probably TASK_ID is also a type that is VxWorks 6.9 specific

File: ACE/ace/OS_NS_signal.inl [hunk 3]
```diff
@@ -15,6 +15,14 @@ kill (pid_t pid, int signum)
   ACE_UNUSED_ARG (pid);
   ACE_UNUSED_ARG (signum);
   ACE_NOTSUP_RETURN (-1);
+#elif defined (ACE_VXWORKS)
+  /*
```
Comment (jwillemsen) [inline ACE/ace/OS_NS_signal.inl]: What when RTP is used instead of kernel mode, at that moment VxWorks does have support for a process

Comment (shuston) [inline ACE/ace/OS_NS_signal.inl]: That's fine - the only difference is the signature for kill() on VxWorks, and it should work regardless. This change can stay unless/until someone with a VxWorks build environment can test it.

File: ACE/ace/OS_NS_Thread.cpp (hunk 2 above)
Comment (shuston) [inline ACE/ace/OS_NS_Thread.cpp]: Probably right... will change this.

File: ACE/ace/OS_NS_Thread.cpp (hunk 1 above)
Comment (shuston) [inline ACE/ace/OS_NS_Thread.cpp]: Ok, will change this.

**Pull request merged**